### PR TITLE
Bundle all components on a single library

### PR DESCRIPTION
**Pull request by mbelluzzo**
And install it.

Headers won't be installed yet because we need to review it first.

As a bonus we got some seconds when making with '-j' :-)
```
[master]$ time cmake .. -DCMAKE_PREFIX_PATH=$HOME/raiz -DWITH_TOOLS=ON -DWITH_GAZEBO=ON -DWITH_VDEBUG=ON -DWITH_SAMPLES=ON
real	0m2.089s
user	0m1.447s
sys	0m0.303s

[master]$ time make
real	0m42.900s
user	0m40.473s
sys	0m2.087s

[master]$ time make -j8
real	0m23.770s
user	0m58.307s
sys	0m2.823s

[master]$ time make -j16
real	0m23.351s
user	0m56.960s
sys	0m2.787s

[libcoav]$ time cmake .. -DCMAKE_PREFIX_PATH=$HOME/raiz -DWITH_TOOLS=ON -DWITH_GAZEBO=ON -DWITH_VDEBUG=ON -DWITH_SAMPLES=ON
real	0m2.054s
user	0m1.440s
sys	0m0.263s

[libcoav]$ time make
real	0m42.929s
user	0m40.600s
sys	0m2.023s

[libcoav]$ time make -j8
real	0m20.273s
user	1m0.007s
sys	0m2.707s

[libcoav]$ time make -j16
real	0m20.570s
user	0m59.423s
sys	0m3.010s
```

### CODE REVIEW

**Comment by mbelluzzo** [comment]
#86

File: CMakeLists.txt [hunk 1]
```diff
@@ -104,13 +104,13 @@ if (${WITH_REALSENSE})
...
-add_subdirectory(detection)
-add_subdirectory(sensors)
-add_subdirectory(vehicles)
 add_subdirectory(modules/mavlink_vehicles)
+add_subdirectory(src/avoidance)
+add_subdirectory(src/common)
+add_subdirectory(src/communication)
+add_subdirectory(src/detection)
+add_subdirectory(src/sensors)
+add_subdirectory(src/vehicles)
```
Comment (rchiossi) [inline CMakeLists.txt]: That's a nice change! Looks much more organized now! =)

File: src/common/CMakeLists.txt [hunk 2]
```diff
@@ -2,8 +2,6 @@ cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
 
 set (sources
   math.cc
-  DepthCamera.cc
-)
+  DepthCamera.cc)
```
Comment (rchiossi) [inline src/common/CMakeLists.txt]: Is that some code style standard for cmake or you just like it in a single line?I noticed you changed this in the entire project.

Comment (mbelluzzo) [inline src/common/CMakeLists.txt]: Just following the same cmake code style that I see being used on other projects. I hope to add code style check on the project soon and probably add checks for the cmake files as well. Actually for long lists I prefer on a new line, but for short lists its indeed uglier compared with same line.

**Pull request merged**